src/skmultiflow/utils/data_structures.py

```diff
@@ -1014,3 +1015,142 @@ def get_info(self):
                ' - n_samples: ' + str(self.n_samples) + \
                ' - max_size: ' + str(self.max_size) + \
                ' - dtype: ' + str(self.dtype)
+
+
+class TimeManager(object):
+    """ TimeManager
+
+    Manage instances that are related to a timestamp and
+    a delay.
+
+    Parameters
+    ----------
+    timestamp: np.datetime64
+        Current timestamp of the stream. This timestamp is always
+        updated as the stream is processed to simulate when
+        the labels will be available for each sample.
+
+    """
+
+    _columns = ["X", "y_real", "y_pred", "arrival_time", "available_time"]
+
+    def __init__(self, timestamp):
+        # get initial timestamp
+        self.timestamp = timestamp
+        # create dataframe to save time data
+        # X = features
+        # y_real = real label
+        # y_pred = predicted label for each model being evaluated
+        # arrival_time = arrival timestamp of the sample
+        # available_time = timestamp when the sample label will be available
+        self.queue = pd.DataFrame(columns=self._columns)
+        # transform queue into datetime if timestamps are not int
+        if isinstance(self.timestamp, int):
+            self.queue['arrival_time'] = pd.to_datetime(self.queue['arrival_time'])
+            self.queue['available_time'] = pd.to_datetime(self.queue['available_time'])
+
+    def _sort_queue(self):
+        # sort values by available_time
+        self.queue = self.queue.sort_values(by='available_time')
+        # reset indexes
+        self.queue = self.queue.reset_index(drop=True)
```

Comment:
Would this be achievable in one step?
```python
self.queue = self.queue.sort_values(by='available_time', ignore_index=True)
```

Reply:
Nice catch! Included this in commit [344fd621b768f0dac966f867b9c3ae19b95d8b67](https://github.com/scikit-multiflow/scikit-multiflow/commit/344fd621b768f0dac966f867b9c3ae19b95d8b67).